internal/agent/list.go

```diff
@@ -12,41 +12,86 @@ import (
 var agentListStyle = lipgloss.NewStyle().Margin(1, 2)
 
 type AgentListModel struct {
-	agentList   list.Model
-	agentLoader tea.Cmd
+	agentList        list.Model
+	agentCurrentPage int
+	agentPerPage     int
+	agentLastPage    int
+	agentsLoading    bool
+	agentLoader      func(int, int) tea.Cmd
 }
 
-func NewAgentList(loader tea.Cmd) AgentListModel {
+func NewAgentList(loader func(int, int) tea.Cmd, page, perpage int) AgentListModel {
 	l := list.New(nil, NewDelegate(), 0, 0)
 	l.Title = "Buildkite Agents"
 
 	return AgentListModel{
-		agentList:   l,
-		agentLoader: loader,
+		agentList:        l,
+		agentCurrentPage: page,
+		agentPerPage:     perpage,
+		agentLoader:      loader,
 	}
 }
 
+func (m *AgentListModel) appendAgents() tea.Cmd {
+	// Set agentsLoading
+	m.agentsLoading = true
+	// Set a status message and start the agentList's spinner
+	startSpiner := m.agentList.StartSpinner()
+	setStatus := m.agentList.NewStatusMessage(("Fetching more agents"))
+	// Fetch and append more agents
+	appendAgents := m.agentLoader(m.agentCurrentPage, m.agentPerPage)
+	return tea.Sequence(appendAgents, tea.Batch(startSpiner, setStatus))
+}
+
 func (m AgentListModel) Init() tea.Cmd {
-	return m.agentLoader
+	return m.agentLoader(m.agentCurrentPage, m.agentPerPage)
```

Comment:
Should this use the appendAgents method instead now? And then perhaps the status message should just be "Loading agents", or it can check what the current page is and change the status message

Reply:
I think thats a nice idea @jradtilbrook - maybe `Loading agents: page X of Y` with combination for using `appendAgents`?

Reply:
I've tweaked the message to read `Loading more agents: page X of Y` (using the current page and last page params returned from the API)

Reply:
![image](https://github.com/buildkite/cli/assets/70425486/c8432ccb-fd46-4ca1-b622-1450fe988591)

Reply:
I wonder if that is a little confusing since this presents an infinite scroll list. Not a big deal though 👍

Reply:
Agreed - we probably can change it if it gets confusing, the actual list is paged also with the amount of items that it returns 🤔